From 4dd1db0d13a2f2623e7aa94d351396b59c98df2e Mon Sep 17 00:00:00 2001
From: Fugang Duan <fugang.duan@nxp.com>
Date: Tue, 29 Aug 2017 10:21:09 +0800
Subject: [PATCH 2/5] hciattach: set flag to enable HCI reset on init

On some qca devices, correct setup of BT+WLAN co-existance requires
HCI reset being set at the start of power on sequence. Sending HCI
resets has no side effect.

Upstream-Status: Inappropriate [enable feature]
Change-Id: I71cb367d10d4d19d82b41af6a4a0b8b2f770f691

Signed-off-by: Rupesh Tatiya <rtatiya@codeaurora.org>
---
 tools/hciattach.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/tools/hciattach.c b/tools/hciattach.c
index 200f7340b..d03ed468f 100644
--- a/tools/hciattach.c
+++ b/tools/hciattach.c
@@ -1158,6 +1158,9 @@ static int init_uart(char *dev, struct uart_t *u, int send_break, int raw)
 	if (u->flags & AMP_DEV)
 		flags |= 1 << HCI_UART_CREATE_AMP;
 
+	if (!strncmp(u->type, "qca", 3))
+		flags |= 1 << HCI_UART_RESET_ON_INIT;
+
 	fd = open(dev, O_RDWR | O_NOCTTY);
 	if (fd < 0) {
 		perror("Can't open serial port");
-- 
2.17.1

